### ISSUE TITLE
Add profiling system for the shock tube problem

### ISSUE THREAD

**Issue by yungyuc** [solvcon/solvcon#135]
The shock tube problem (`modmesh.onedim.euler1d.ShockTube`) is a classical compressible flow problem.  It uses a numerical solver along with an analytical solution.

The numerical solver is supposed to be high-performance but it is not yet optimized.  Before optimization, profiling should be done.  We do not yet have a useable profiling system in modmesh, but there is a prototype: https://github.com/solvcon/modmesh/blob/master/cpp/modmesh/profile.hpp .

This issue is to track the augmentation of the prototype to a useable profiling system.  It does not need to be comprehensive, but should help understand the performance of the shock tube problem.

**Comment by yungyuc** [comment]
To have a basic understandings of the shock tube problem, the notes in https://github.com/solvcon/mmnote/ may be helpful.

**Comment by tai271828** [comment]
I will help profiling on Ubuntu Linux.

**Comment by tai271828** [comment]
Since pull request #190 is landed, I think my next step will try to make the profiling data able to be exported to a file, which could be consumed by a GUI-based reporter, dashboard, or something.

easy_profiler mentioned in [this thread](https://github.com/solvcon/modmesh/pull/188#discussion_r1056893194) has a nice GUI application to represent its profiling data. The application is written in Qt5, so I believe we can be enlightened a lot by its code when writing a Qt6-based dashboard for our own profiling data.

**Comment by tai271828** [comment]
> Since pull request #190 is landed, I think my next step will try to make the profiling data able to be exported to a file, which could be consumed by a GUI-based reporter, dashboard, or something.
> 
> easy_profiler mentioned in [this thread](https://github.com/solvcon/modmesh/pull/188#discussion_r1056893194) has a nice GUI application to represent its profiling data. The application is written in Qt5, so I believe we can be enlightened a lot by its code when writing a Qt6-based dashboard for our own profiling data.

Besides, if the data could be consumed by the other application, that means it will be time for us to put profiling benchmark in our CI.

**Comment by yungyuc** [comment]
Yes, I think it's a great idea to steal profling visualization code from easy profiler, and probably also the profile data file format.

Profiling in CI is tricky because the CI servers usually are noisy.  We should bring it to discord and use more discussions for it.

**Comment by yungyuc** [comment]
Clear assignees for lack of activities.

**Comment by tigercosmos** [comment]
the profile.hpp prototype grew into a full system under cpp/solvcon/profiling/ (CallProfiler, RadixTree, SerializableProfiler) with a pilot GUI, and the shock-tube Euler1DCore is instrumented (SOLVCON_PROFILE_SCOPE, def_timed).
